src/Helpers/StacktraceExtractor.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace AG\ElasticApmLaravel\Helpers;
+
+use Illuminate\Config\Repository as Config;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Str;
+
+class StacktraceExtractor
+{
+    public static function getStacktrace(Config $config): array
+    {
+        $stackTrace = self::stripVendorTraces(
+            collect(
+                debug_backtrace(DEBUG_BACKTRACE_PROVIDE_OBJECT, $config->get('elastic-apm-laravel.spans.backtraceDepth', 0))
+            )
+        );
+
+        return $stackTrace->map(function ($trace) use ($config) {
+            $sourceCode = self::getSourceCode($trace, $config);
+
+            return [
+                'function' => Arr::get($trace, 'class') . Arr::get($trace, 'type') . Arr::get($trace, 'function'),
+                'abs_path' => Arr::get($trace, 'file'),
+                'filename' => basename(Arr::get($trace, 'file')),
+                'lineno' => Arr::get($trace, 'line', 0),
+                'library_frame' => false,
+                'pre_context' => optional($sourceCode->get('pre_context'))->toArray(),
+                'context_line' => optional($sourceCode->get('context_line'))->first(),
+                'post_context' => optional($sourceCode->get('post_context'))->toArray(),
+            ];
+        })->values()->toArray();
+    }
+
+    private static function getSourceCode(array $stackTrace, Config $config): Collection
+    {
+        if (false === $config->get('elastic-apm-laravel.spans.renderSource', true)) {
```

Comment:
Can you move this to the beginning of `getStacktrace` method? This way we avoid unnecessary loops.